pandas/tests/test_window.py

```diff
@@ -289,6 +290,200 @@ def test_deprecations(self):
+
+    def get_expects(self):
+        expects = {
+            'sr1': {
+                'count': Series([1, 2, 2, 2, 2]),
+                'max': Series([self.naval, 1, 2, 3, 4]),
+                'min': Series([self.naval, 0, 1, 2, 3]),
+                'sum': Series([self.naval, 1, 3, 5, 7]),
+                'mean': Series([self.naval, .5, 1.5, 2.5, 3.5]),
+                'std': Series([self.naval] + [np.sqrt(.5)] * 4),
+                'var': Series([self.naval, .5, .5, .5, .5]),
+                'median': Series([self.naval, .5, 1.5, 2.5, 3.5])
+            },
+            'sr2': {
+                'count': Series([1, 2, 2, 2, 2]),
+                'max': Series([self.naval, 10, 8, 6, 4]),
+                'min': Series([self.naval, 8, 6, 4, 2]),
+                'sum': Series([self.naval, 18, 14, 10, 6]),
+                'mean': Series([self.naval, 9, 7, 5, 3]),
+                'std': Series([self.naval] + [np.sqrt(2)] * 4),
+                'var': Series([self.naval, 2, 2, 2, 2]),
+                'median': Series([self.naval, 9, 7, 5, 3])
+            },
+            'df': {
+                'count': DataFrame({0: Series([1, 2, 2, 2, 2]),
+                                    1: Series([1, 2, 2, 2, 2])}),
+                'max': DataFrame({0: Series([self.naval, 2, 4, 6, 8]),
+                                  1: Series([self.naval, 3, 5, 7, 9])}),
+                'min': DataFrame({0: Series([self.naval, 0, 2, 4, 6]),
+                                  1: Series([self.naval, 1, 3, 5, 7])}),
+                'sum': DataFrame({0: Series([self.naval, 2, 6, 10, 14]),
+                                  1: Series([self.naval, 4, 8, 12, 16])}),
+                'mean': DataFrame({0: Series([self.naval, 1, 3, 5, 7]),
+                                  1: Series([self.naval, 2, 4, 6, 8])}),
+                'std': DataFrame({0: Series([self.naval] + [np.sqrt(2)] * 4),
+                                  1: Series([self.naval] + [np.sqrt(2)] * 4)}),
+                'var': DataFrame({0: Series([self.naval, 2, 2, 2, 2]),
+                                  1: Series([self.naval, 2, 2, 2, 2])}),
+                'median': DataFrame({0: Series([self.naval, 1, 3, 5, 7]),
+                                     1: Series([self.naval, 2, 4, 6, 8])}),
+            }
+        }
+        return expects
+
+    def _create_dtype_data(self, dtype):
+        sr1 = Series(range(5), dtype=dtype)
+        sr2 = Series(range(10, 0, -2), dtype=dtype)
+
+        data = {
+            'sr1': sr1,
+            'sr2': sr2
+        }
+        if self.include_df:
+            df = DataFrame(np.arange(10).reshape((5, 2)), dtype=dtype)
+            data['df'] = df
+
+        return data
+
+    def _create_data(self):
+        super(TestDtype, self)._create_data()
+        self.data = self._create_dtype_data(self.dtype)
+        self.expects = self.get_expects()
+
+    def setUp(self):
+        self._create_data()
+
+    def _cast_result(self, result, from_dtype, to_dtype):
```

Comment:
override ``create_data_types`` in Testdatetimelike. ``setUp`` should NEVER raise, the point is its a known starting point. Really try to make this as simple as possible. I am happy for you to skip ALL Datetimelike test ATM.